guardian/cdk
# Add GuArnParameter with ARN regex allowed values

## What does this change?

This PR adds a `GuArnParameter`. This allows users of the library to add parameters to their stacks that are constrained by a regex enforcing them to pass ARNs compliant with [AWS's documentation](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).

The `AllowedValues` regex has been tested and appears to be working.

## Does this change require changes to existing projects or CDK CLI?

No.

## How to test

Paste the regex in `allowedValues` into a [regex tester](https://regex101.com/), then paste some valid (and invalid, if desired) ARNs against it and see it successfully validate the correct strings.

## How can we measure success?

Future projects can create parameters that only allow ARN values.

## Screenshots

Valid ARN compositions:
![image](https://user-images.githubusercontent.com/25747336/101520120-f048de00-397b-11eb-8b95-dd9cf1c57182.png)
